src/metabase/db/custom_migrations.clj

```diff
@@ -1743,3 +1747,131 @@
 (define-reversible-migration AddStageNumberToParameterMappingTargets
   (add-stage-numbers-to-parameter-mapping-targets)
   (remove-stage-numbers-from-parameter-mapping-targets))
+
+(defn- dimension?
+  [x]
+  (and (vector? x)
+       (< 1 (count x) 4)
+       (= (get x 0) "dimension")))
+
+(defn- update-parameter-mapping-target-dimension
+  [parameter-mapping-entry update-fn]
+  (let [parameter-mapping (val parameter-mapping-entry)
+        target (:target parameter-mapping)
+        dimension (:dimension target)
+        new-dimension (cond-> dimension
+                        (and (= (:type target) "dimension")
+                             (dimension? dimension))
+                        update-fn)]
+    (if (not= new-dimension dimension)
+      (let [new-dimension-str (json/generate-string new-dimension)
+            new-dimension-key (keyword new-dimension-str)]
+        [new-dimension-key (-> parameter-mapping
+                               (assoc :id new-dimension-str)
+                               (update :target assoc :id new-dimension-str :dimension new-dimension))])
+      parameter-mapping-entry)))
+
+(defn- update-column-settings-target-dimensions
+  "This can update any map containing :click_behavior and is called with visualization_settings too."
+  [column-settings click-behavior->update-fn]
+  (let [click-behavior (:click_behavior column-settings)
+        update-fn (click-behavior->update-fn click-behavior)]
+    (cond-> column-settings
+      (and update-fn
+           (= (:linkType click-behavior) "question")
+           (-> click-behavior :parameterMapping map?))
+      (update-in [:click_behavior :parameterMapping]
+                 (fn [mappings]
+                   (into {}
+                         (map #(update-parameter-mapping-target-dimension % update-fn))
+                         mappings))))))
+
+(defn- update-viz-settings-target-dimensions
+  [viz-settings click-behavior->update-fn]
+  (cond-> viz-settings
+    (-> viz-settings :column_settings map?)
+    (update :column_settings
+            update-vals
+            #(update-column-settings-target-dimensions % click-behavior->update-fn))))
```

Comment:
Can this be `m/update-existing` instead?

Reply:
I prefer making sure here that the value is indeed a map. I don't want to mess with anything else.